docs/advanced/integration_tests.md

```diff
@@ -0,0 +1,132 @@
+}
+```
+
+The `Test_ToEuro` method makes a GET call to an endpoint, in order to receive back the euro exchange value for 10 GBP. Lastly, it checks if the received value is 8.0.
+
+In order to make this GET call `Test_ToEuro` does not need to concern itself about the IP on which the exchange application is running, nor its port. All `Test_ToEuro` needs to know is the URL path of that endpoint, as the _client_ object does the rest.
+
+This _client_ object is provided out-of-the box by Gosoline, whenever a test is defined like this:
+
+[embedmd]:# (../../examples/getting_started/integration/api_test.go /func \(s \*ApiTestSuite\) SetupSuite/ /return definer.ApiDefiner\n}/)
+```go
+func (s *ApiTestSuite) SetupSuite() []suite.Option {
+	return []suite.Option{
+		suite.WithLogLevel("info"),
+		suite.WithConfigFile("../api/config.dist.yml"),
+		suite.WithFixtures(fixtureSets),
+		suite.WithClockProvider(s.clock),
+	}
+}
+
+func (s *ApiTestSuite) SetupApiDefinitions() apiserver.Definer {
+	return definer.ApiDefiner
+}
+```
+
+Implementing _SetupApiDefinitions_ when configuring your test will inform Gosoline that an API is being tested, thus the _client_ object will contain meaningful data. The `SetupSuite` method is shown to illustrate how similar is the configuration for all three types of tests.
+
+Forgeting to implement _SetupApiDefinitions_, and the trying to run an _resty.Client_ test case, will result in an error reminding you that `the suite has to implement the TestingSuiteApiDefinitionsAware interface`.
+
+### ApiServer integration tests
+
+The _client_ object is permissive, allowing access to the request itself. A simpler way to test an API is provided by Gosoline's _ApiServerTestCase_. These type of tests also issue HTTP calls and compare their responses against a predefined value, but they need to follow a predefined structure. Below we can see how the _resty client_ test above would look like as an _ApiServerTestCase_:
+
+[embedmd]:# (../../examples/getting_started/integration/api_test.go /func \(s \*ApiTestSuite\) Test_Euro/ /\n}/)
+```go
+func (s *ApiTestSuite) Test_Euro() *suite.ApiServerTestCase {
+	return &suite.ApiServerTestCase{
+		Method:             http.MethodGet,
+		Url:                "/euro/10/GBP",
+		Headers:            map[string]string{},
+		ExpectedStatusCode: http.StatusOK,
+		Assert: func(response *resty.Response) error {
+			s.Equal(http.StatusOK, response.StatusCode())
+
+			result, err := strconv.ParseFloat(string(response.Body()), 64)
+			s.NoError(err)
+			s.Equal(8.0, result)
+
+			return nil
+		},
+	}
+}
+```
+
+Notice the predefined structure of an _ApiServerTestCase_: it is an object that has fields for all the information needed in performing an HTTP call, including a method that tests the result for correctness.
+
+In the example from `examples/getting_started/integration/api_test.go` we can see that both types of test cases (_resty.Client_ and _ApiServerTestCase_) share the same `SetupSuite` method.
```

Comment:
I would simplify that to something similar to "you can use both in the same test suite"